File: QuickBuy/QuickBuy.Dominio/Entidades/Pedido.cs
using QuickBuy.Dominio.ObjetoValor;
using System;
using System.Collections.Generic;
using System.Linq;

namespace QuickBuy.Dominio.Entidades
{
    public class Pedido : Entidade
    {
        public int Id { get; set; }

        public DateTime DataPedido { get; set; }

        public DateTime DataPrevisaoEntrega { get; set; }

        public int UsuarioId { get; set; }

        public virtual Usuario Usuario { get; set; }

        public int EnderecoId { get; set; }

        public virtual Endereco Endereco { get; set; }

        public int FormaPagamentoId { get; set; }

        public virtual FormaPagamento FormaPagamento { get; set; }

        public virtual ICollection<ItemPedido> ItensPedidos { get; set; }

        public override void ValidaDados()
        {
            if(!ItensPedidos.Any())
            {
                AdicionarMsgCritica("Pedido não pode ficar sem item de pedido.");
            }
        }
    }
}

File: QuickBuy/QuickBuy.Web/ClientApp/src/app/model/pedido.ts
import { ItemPedido } from "./itemPedido";

export class Pedido
{
  id: number;
  dataPedido: Date;
  dataPrevisaoEntrega: Date;
  usuarioId: number;
  enderecoId: number;
  formaPagamentoId: number;
  valorTotal: number;
  itensPedido: ItemPedido[];

  constructor() {
    this.dataPedido = new Date();
    this.itensPedido = [];
  }
}
